### ISSUE TITLE
Focused view missing for text editor

### ISSUE THREAD

**Issue by AndreasArvidsson** [microsoft/vscode#192534]
Huge fan of `focusedView` for the `window.title` setting as an accessibility user. The problem is that I really need to know when the text editor is focused or not. Currently I can't tell if I'm in the editor or the search widget for example. Could we add `Text Editor` as a view name in `focusedView`?

edit:
As a clarification this would be only when the focus is actually on the text editor document itself. No search or rename widgets. Preferably those would have their own names as well, but are not as important.

**Comment by VSCodeTriageBot** [comment]
This feature request is now a candidate for our backlog. The community has 60 days to [upvote](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) the issue. If it receives 20 upvotes we will move it to our backlog. If not, we will close it. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by VSCodeTriageBot** [comment]
:slightly_smiling_face: This feature request received a sufficient number of community upvotes and we moved it to our backlog. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by meganrogge** [comment]
If you use `focusedView` and `activeEditor` in the window title, does that work for you?

**Comment by AndreasArvidsson** [comment]
Unfortunately no. When in the search widget `focusedView` is missing and `activeEditorShort` shows the filename. This is the same as if the text document was actually focused. I need to know if focus is actually on the text document and not search/rename widgets/popups. Something similar to the keybinding `"when": "editorTextFocus"`, but in the window title instead.

**Comment by meganrogge** [comment]
Do you mean when you focus the search widget in the editor? Or the search input in the search view?

**Comment by AndreasArvidsson** [comment]
Yes the search widget in the editor. I need to be able to detect this. `editorTextFocus` works, but is not accesible through the title.

**Comment by meganrogge** [comment]
The editor is not a view and thus, I am not sure we should do this. @sandy081 what do you think?

**Comment by meganrogge** [comment]
We could consider adding another variable, `${editorTextFocused}` or something though

**Comment by AndreasArvidsson** [comment]
`${editorTextFocused}` would work great for me!

**Comment by meganrogge** [comment]
@bpasero what do you think of this? Would we go with `Editor Focused` when true or `True`... 🤔

**Comment by bpasero** [comment]
I would not have another one but use `focusedView`. Can we rename it so that it covers views and parts? It is a bit weird though to not have it announce the editor when focus is in the search field. Should we then do the same in the views as well?

**Comment by meganrogge** [comment]
The request is for it to specifically not say the `focusedView` is editor when it's in the editor's find widget. 

When the terminal is the focused view, it remains that even when using the find widget. That makes sense to me as it's still the focused view. 

Thus, I feel it's best to have a separate one, so we aren't doing different things / the definition of focused view doesn't become quite confusing.

**Comment by bpasero** [comment]
Then go ahead and add a new variable as you see fit for this use case, I think there is no limits to what we can have there in terms of variables in that setting.

**Comment by meganrogge** [comment]
This is a bit more involved than I had imagined. We currently don't have an event which tracks when a find widget is shown /hidden. We'd need that in order to schedule an update of the window title.

**Comment by bpasero** [comment]
I think the monaco code editor exposes 2 events: `onDidFocusEditorText` and `onDidFocusEditorWidget` that would need to be brought up to the title control I guess.

**Comment by meganrogge** [comment]
I agree with you now that I've looked into this more closely. I think it should be included in the `focusedView` variable and we should call that something different.

**Comment by meganrogge** [comment]
interestingly, `onDidFocusEditorWidget` was not fired when I expected it to be. `onDidFocusEditorText` and `onDidBlurEditorText` work well though 👍🏼 thanks.

**Comment by meganrogge** [comment]
I don't want to break peoples' existing setups, so maybe we should stick with `focusedView`

**Comment by meganrogge** [comment]
To verify:

- add `${focusedView}` to `window.title`
- Focus different views (for example, the output channel, terminal, SCM) 
- Focus a text editor. You should see `Text Editor` in the window title.
- Use the find widget in a text editor. You should not see `Text Editor` in the window title.
